arches_lingo/src/arches_lingo/App.vue

```diff
@@ -22,32 +27,36 @@ provide(userKey, { user, setUser });
 
 const router = useRouter();
 const route = useRoute();
+const toast = useToast();
+const { $gettext } = useGettext();
 
-const isAuthenticated = computed(() => {
-    return user.value && user.value.username !== "anonymous";
-});
+router.beforeEach(async (to, _from, next) => {
+    try {
+        const userData = await fetchUser();
+        setUser(userData);
```

Comment:
Great solution, let's just only fetch the data if it's missing -- my idea was to fetch user details on refresh, not on in-app route-navigation.

```suggestion
        let userData = user.value;
        if (!userData) {
            userData = await fetchUser();
            setUser(userData);
        }
```

Reply:
++